stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/DateFormatter.kt

```diff
@@ -61,75 +59,123 @@ public interface DateFormatter {
+    /**
+     * Formats the given time as a String.
+     *
+     * @param date The [Date] object to format as a String.
+     * @return The formatted time string.
+     */
+    override fun formatTime(date: Date?): String {
+        date ?: return ""
+
+        val dateFormat = if (dateContext.is24Hour()) timeFormatter24h else timeFormatter12h
+        return dateFormat.format(date)
+    }
+
+    /**
+     * Checks if the supplied day is today.
+     *
+     * @return true if the date is today.
+     */
+    private fun Date.isToday(): Boolean {
+        val calendar1 = Calendar.getInstance().also { it.time = dateContext.now() }
+        val calendar2 = Calendar.getInstance().also { it.time = this }
+
+        return (calendar1[Calendar.YEAR] == calendar2[Calendar.YEAR]) &&
+            calendar1[Calendar.DAY_OF_YEAR] == calendar2[Calendar.DAY_OF_YEAR]
+    }
+
+    /**
+     * Checks if the supplied date is yesterday.
+     *
+     * @return True if the date is yesterday.
+     */
+    private fun Date.isYesterday(): Boolean {
+        return Date(time + DateUtils.DAY_IN_MILLIS).isToday()
     }
 
-    private fun LocalDate.isToday(): Boolean {
-        return this == dateContext.now()
+    /**
+     * Checks if the supplied date is within last week.
+     *
+     * @return True is the date is within last week.
+     */
+    private fun Date.isWithinLastWeek(): Boolean {
+        return isWithinDays(DAYS_IN_WEEK - 1)
     }
 
-    private fun LocalDate.isYesterday(): Boolean {
-        return this == dateContext.now().minusDays(1)
+    /**
+     * Checks if the supplied date is before today and within a number of days in the past.
+     *
+     * @param days The number of days before the current date.
+     * @return True is the date is within x days in the past.
+     */
+    private fun Date.isWithinDays(days: Int): Boolean {
+        val calendar: Calendar = Calendar.getInstance().also { it.time = this }
+
+        val currentDate = dateContext.now()
+        val start: Calendar = Calendar.getInstance().also {
+            it.time = currentDate
+            it.add(Calendar.DAY_OF_YEAR, -days)
+        }
+        val end: Calendar = Calendar.getInstance().also { it.time = currentDate }
+
+        return calendar.isBeforeDay(end) && !calendar.isBeforeDay(start)
     }
 
-    private fun LocalDate.isWithinLastWeek(): Boolean {
-        return this > dateContext.now().minusDays(DAYS_IN_A_WEEK)
+    /**
+     * Checks if the calendar date is before another calendar date ignoring time.
+     *
+     * @return True if the calendar date is before another calendar date ignoring time.
+     */
+    private fun Calendar.isBeforeDay(calendar: Calendar): Boolean {
+        if (this[Calendar.YEAR] < calendar[Calendar.YEAR]) return true
+        if (this[Calendar.YEAR] > calendar[Calendar.YEAR]) return false
+        return this[Calendar.DAY_OF_YEAR] < calendar[Calendar.DAY_OF_YEAR]
```

Comment:
```suggestion
        return when {
            this[Calendar.YEAR] < calendar[Calendar.YEAR] -> true
            this[Calendar.YEAR] > calendar[Calendar.YEAR] -> false
            else -> this[Calendar.DAY_OF_YEAR] < calendar[Calendar.DAY_OF_YEAR]
        }
```

Just my preference, up to you.